include/ginkgo/core/base/mpi.hpp

```diff
@@ -501,6 +549,53 @@ class communicator {
      */
     bool operator!=(const communicator& rhs) const { return !(*this == rhs); }
 
+    /**
+     * Checks if the rhs communicator is identical to this communicator.
+     *
+     * @note If MPI_COMM_NULL is the underlying MPI_COMM of this, then the
+     *       communicator is equal to rhs, iff MPI_COMM_NULL is also the
+     *       underlying MPI_COMM of rhs.
+     *
+     * @return  true if rhs is identical to this
+     */
+    bool is_identical(const communicator& rhs) const
+    {
+        if (get() == nullptr || rhs.get() == nullptr) {
+            return get() == rhs.get();
+        }
+        if (get() == MPI_COMM_NULL || rhs.get() == MPI_COMM_NULL) {
+            return get() == rhs.get();
+        }
+        int flag;
+        GKO_ASSERT_NO_MPI_ERRORS(MPI_Comm_compare(get(), rhs.get(), &flag));
+        return flag == MPI_IDENT;
+    }
+
+    /**
+     * Checks if the rhs communicator is congruent to this communicator.
+     *
+     * Congruent communicators are defined as having identical rank members and
+     * rank ordering.
+     *
+     * @note If MPI_COMM_NULL is the underlying MPI_COMM of this, then the
+     *       communicator is congruent to rhs, iff MPI_COMM_NULL is also the
+     *       underlying MPI_COMM of rhs.
+     *
+     * @return  true if rhs is congruent to this
+     */
+    bool is_congruent(const communicator& rhs) const
+    {
+        if (!get() || !rhs.get()) {
+            return get() == rhs.get();
+        }
+        if (get() == MPI_COMM_NULL || rhs.get() == MPI_COMM_NULL) {
+            return get() == rhs.get();
+        }
+        int flag;
+        GKO_ASSERT_NO_MPI_ERRORS(MPI_Comm_compare(get(), rhs.get(), &flag));
+        return flag == MPI_CONGRUENT;
```

Comment:
identical should be also congruent, right?
In usecases, should it check `flag == MPI_CONGRUENT || flag == MPI_IDENT`?

Reply:
I guess so, but I would still leave it as it is. This function should be considered quite advanced, so I think it would be simpler to just stick with the MPI standard.